presto-spi/src/main/java/com/facebook/presto/spi/type/JsonSerde.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class JsonSerde
+{
+    private JsonSerde() {}
+
+    public static boolean isValidJsonObjectKeyType(Type type)
+    {
+        return isValidJsonObjectKeyType(type.getTypeSignature());
+    }
+
+    public static boolean isValidJsonObjectKeyType(TypeSignature typeSignature)
+    {
+        String baseType = typeSignature.getBase();
+        return baseType.equals(StandardTypes.BOOLEAN) ||
+                baseType.equals(StandardTypes.BIGINT) ||
+                baseType.equals(StandardTypes.DOUBLE) ||
+                baseType.equals(StandardTypes.TIMESTAMP) ||
+                baseType.equals(StandardTypes.VARCHAR);
+    }
+
+    public static class MapEntry<K, V>
```

Comment:
Using this class will allow us to upgrade to the newer Jackson versions that handle maps differently.